### ISSUE TITLE
Deploy the farming smart contract for Nabla

### ISSUE THREAD

**Issue by prayagd** [pendulum-chain/pendulum#373]
## Context
In order to enable farming and distribute rewards to the Nabla LP's. We want to deploy the farming smart contract on Pendulum. This is the follow up ticket of the research done on this smart contract whether it is compatible with Pendulum, results can be found [here](https://www.notion.so/satoshipay/23-10-25-Smart-Contract-for-Farming-Rewards-467ea92c0b5f4e49987fe74266af9a11?pvs=4). 

tl;dr - 
Can it be compiled to WASM? 
It is. With only a minor modification. It was compiled and deployed to foucoco-standalone using wasm-deploy. Along with the `SafeERC20` contract which is already implemented by Nabla

## Requirement

- [ ] Deploy the farming smart contract Foucoco
- [ ] Test the smart contract
- [ ] Deploy it on Pendulum

**Comment by prayagd** [comment]
@ebma @gianfra-t Please check and let me know if anything missing. Also i assume you would need certain parameters that needs to be configures in this smart contract, if you can help me with what are those? i can get this ticket ready for execution.

**Comment by prayagd** [comment]
@ebma ok, will get you these parameters but would this be required to deploy it on foucoco and test the rewards? If not can we progress on foucoco then?

**Comment by gianfra-t** [comment]
Yes those things we need to pass when creating the contract. But I imagine we could deploy in foucoco with some mock values to test the contract.

**Comment by prayagd** [comment]
Ok, putting this on hold i until i get the production values

**Comment by TorstenStueber** [comment]
@prayagd I would continue to be agile and not prioritize this ticket. No urgency for this at the moment.

**Comment by TorstenStueber** [comment]
@pendulum-chain/product Ping

**Comment by prayagd** [comment]
@TorstenStueber agree and hence put this on hold

**Comment by vadaynujra** [comment]
@prayagd in light of the off ramp feature we're building, how do we plan to get the initial liquidity to Nabla for the stablecoins? When would an LP incentives feature be required on Nabla?

**Comment by TorstenStueber** [comment]
I think we should plan this around how we are going to use Nabla or its different instances in the next couple of months.

**Comment by vadaynujra** [comment]
If we want organic liquidity, we would need to consider incentivising LPers, at least initially, when the swap traction is low. 
We could do this manually but on-chain rewards would just be simpler, though implementing it might not be. 
@TorstenStueber what would this kind of plan look like?

**Comment by TorstenStueber** [comment]
I think we should not make these assumptions preemptively before we actually see Nabla in action and observe pain points. Could very well be that we find another solution for liquidity where we don't need to incentivize through farming.

Considering the effort that went into Nabla, I think that this would be quite some effort.

**Comment by TorstenStueber** [comment]
@pendulum-chain/product let's move to the icebox or close.

**Comment by prayagd** [comment]
I would suggest icebox
